packages/compartment-mapper/src/import-archive.js

```diff
@@ -101,6 +144,20 @@ export const parseArchive = async (archiveBytes, archiveLocation) => {
   // TODO validate compartmentMap instead of leaning hard on the above type
   // assertion.
 
+  if (expectedSha512 !== undefined) {
```

Comment:
Is it possible to move the hash-equals-expected test earlier, just after we get `compartmentMapBytes` and before we do anything else with it? That would be the most hygenic.

Reply:
Sure. We can move the check before parsing the content of the compartment map.